js/PulsarFormComponent.js

```diff
@@ -84,33 +87,54 @@ PulsarFormComponent.prototype.initTimePickers = function () {
  */
 PulsarFormComponent.prototype.initDatePickers = function () {
     const datepickers = this.$html.find('[data-datepicker="true"]');
-    let defaultDateFormat = 'DD/MM/YYYY';
 
     datepickers.each((index, element) => {
-        let dateFormat = element.getAttribute('data-format');
+        let $element = $(element);
+        let dateFormat = 'DD/MM/YYYY';
+        let formatKey = element.getAttribute('data-format');
+        let locale = element.getAttribute('data-locale');
 
-        // Check if data-format attribute exists and lowercase it
-        // to eliminate different styles of writing issues
-        if (dateFormat !== null) {
-            dateFormat = dateFormat.toLowerCase();
+        if (typeof locale === 'string') {
+            dateFormat = moment.localeData(locale).longDateFormat('L');
         }
 
-        switch (dateFormat) {
-            case 'us':
-                defaultDateFormat = 'MM/DD/YYYY';
-                break;
-            case 'reverse':
-                defaultDateFormat = 'YYYY/MM/DD';
-                break;
-            default:
-                defaultDateFormat = 'DD/MM/YYYY';
+        if (typeof formatKey === 'string') {
+            formatKey = formatKey.toLowerCase();
+
+            switch (formatKey) {
+                case 'us':
+                    dateFormat = 'MM/DD/YYYY';
+                    break;
+                case 'reverse':
+                    dateFormat = 'YYYY/MM/DD';
+                    break;
+            }
         }
 
         // Initialize pikaday with the correct date format
-        $(element).pikaday({ format: defaultDateFormat });
+        $element.pikaday({ format: dateFormat });
 
         // Initialize placeholder attribute based on the date format
-        $(element).attr('placeholder', defaultDateFormat.toLowerCase());
+        $element.attr('placeholder', dateFormat.toLowerCase());
+
+        if (element.hasAttribute('data-iso') && element.hasAttribute('name')) {
+            let elementName = $element.attr('name');
+            let hiddenElement = $('<input>', {'type': 'hidden', 'name': elementName});
```

Comment:
The hidden element should be added in the Twig. The current implementation has issues when passing a value into the HTML as a default.